alibi_detect/utils/pytorch/distance.py

```diff
@@ -93,8 +115,48 @@ def batch_compute_kernel_matrix(
     return k_mat
 
 
-def mmd2_from_kernel_matrix(kernel_mat: torch.Tensor, m: int, permute: bool = False,
-                            zero_diag: bool = True) -> torch.Tensor:
+def linear_mmd2(
+    x: torch.Tensor,
+    y: torch.Tensor,
+    kernel: Callable
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    Compute maximum mean discrepancy (MMD^2) between 2 samples x and y with the
+    linear-time estimator.
+
+    Parameters
+    ----------
+    x
+        Batch of instances of shape [Nx, features].
+    y
+        Batch of instances of shape [Ny, features].
+    kernel
+        Kernel function.
+    Returns
+    -------
+    MMD^2 between the samples.
+    """
+    n = np.shape(x)[0]
+    m = np.shape(y)[0]
+    if n != m:
+        raise RuntimeError("Linear-time estimator requires equal size samples")
+    k_xx = kernel(x=x[0::2, :], y=x[1::2, :], pairwise=False)
+    k_yy = kernel(x=y[0::2, :], y=y[1::2, :], pairwise=False)
+    k_xy = kernel(x=x[0::2, :], y=y[1::2, :], pairwise=False)
+    k_yz = kernel(x=y[0::2, :], y=x[1::2, :], pairwise=False)
```

Comment:
Is `k_yz` the paper notation? B/c it might be easier to follow by just calling it `k_yx`.

Reply:
Typo, thanks for noticing, fixed.